Makefile

```diff
@@ -7,16 +7,13 @@ SHELL = /bin/bash
 # Build options can either be changed by modifying the makefile, or by building with 'make SETTING=value'
 
 # If COMPARE is 1, check the output md5sum after building
-COMPARE ?= 1
+COMPARE := 1
 # If NON_MATCHING is 1, define the NON_MATCHING C flag when building
-NON_MATCHING ?= 0
+NON_MATCHING := 0
 # If ORIG_COMPILER is 1, compile with QEMU_IRIX and the original compiler
-ORIG_COMPILER ?= 0
+ORIG_COMPILER := 0
 # If COMPILER is "gcc", compile with GCC instead of IDO.
-COMPILER ?= ido
-
-CFLAGS ?=
-CPPFLAGS ?=
```

Comment:
There are `CFLAGS +=` later so I think it'd be cleaner (more obvious to the reader) to initialize them somewhere

Not initializing at all is also the same as `?=`

And finally CFLAGS and CPPFLAGS are standard variables in Make which may have default values or may be part of the environment per user preference

So overall it'd just be better to initialize them, at least for clarity

Reply:
Okay I've reverted that change. Although as I mentioned in the PR description, for environment vs command-line variables is not preference for CFLAGS and CPPFLAGS, it has a functional difference

Reply:
Sorry I don't understand what you mean

Why not have
```
CFLAGS :=
CPPFLAGS :=
```
?

Reply:
Because then I can't write, e.g.

```
CFLAGS=-show make build/src/code/z_collision_check.o
```

to prepend the flag `-show` to the C compiler command. If I instead write

```
make CFLAGS=-show build/src/code/z_collision_check.o
```

then the C flags that are carefully crafted by the Makefile are completely overridden, and it doesn't compile. For a single file perhaps I could copy all the CFLAGS into my command-line instead, but that doesn't work if I want to compile multiple files which might have different flags set by the Makefile.

(I've actually been doing this to debug bss ordering btw, it's not just a theoretical thing)

Reply:
ah I see, okay. I guess the spirit of make is clashing against our "we need specific flags to make OK builds" use case :sweat_smile: 
I guess it's fine to let these be then.

and to be clear afaik
```
SOMEVAR ?=
SOMEVAR += blabla
```
and
```
SOMEVAR += blabla
```
behave exactly the same in make

Reply:
Right, that's why I removed the 

```
CFLAGS ?=
CPPFLAGS ?=
```

lines earlier. Should I un-revert that, or is it clearer the way it is now?

Reply:
I think it's better to keep the empty `?=`s, it's less confusing for reading the makefile

Sorry for the confusion lol